src/dodal/devices/thawer.py

```diff
@@ -1,15 +1,52 @@
+from asyncio import Task, create_task, sleep
 from enum import Enum
 
-from ophyd_async.core import StandardReadable
+from bluesky.protocols import Stoppable
+from ophyd_async.core import (
+    AsyncStatus,
+    Device,
+    SignalRW,
+    StandardReadable,
+)
 from ophyd_async.epics.signal import epics_signal_rw
 
 
+class ThawingException(Exception):
+    pass
+
+
 class ThawerStates(str, Enum):
     OFF = "Off"
     ON = "On"
 
 
-class Thawer(StandardReadable):
+class ThawingTimer(Device):
+    def __init__(self, control_signal: SignalRW[ThawerStates]) -> None:
+        super().__init__("thawing_timer")
+        self._control_signal = control_signal
+        self._thawing_task: Task | None = None
+
+    @AsyncStatus.wrap
+    async def set(self, time_to_thaw_for: float):
+        await self._control_signal.set(ThawerStates.ON)
+        if self._thawing_task and not self._thawing_task.done():
+            raise ThawingException("Thawing task already in progress")
+        self._thawing_task = create_task(sleep(time_to_thaw_for))
+        try:
+            await self._thawing_task
+        finally:
+            await self._control_signal.set(ThawerStates.OFF)
+
+    async def stop(self):
+        if self._thawing_task:
+            self._thawing_task.cancel()
+
+
+class Thawer(StandardReadable, Stoppable):
     def __init__(self, prefix: str, name: str = "") -> None:
         self.control = epics_signal_rw(ThawerStates, prefix + ":CTRL")
+        self.thaw_for_time_s = ThawingTimer(self.control)
         super().__init__(name)
+
+    async def stop(self):
```

Comment:
Should: It's potentially misleading that `thawer.stop()` doesn't actually stop the thawer, it just cancels the `thaw_for_time_s` function. I think calling this should also actively set the control PV to OFF

Reply:
Agree, I think we should have `cancel_if_thawing_on_timer()` and `stop()` be separate things